astropy/time/tests/test_custom_formats.py

```diff
@@ -127,3 +135,56 @@ def value(self):
 
     finally:
         Time.FORMATS.pop("sort", None)
+
+
+def test_mjd_longdouble_preserves_precision(custom_format_name):
+    class CustomMJD(TimeFormat):
+        name = custom_format_name
+
+        def _check_val_type(self, val, val2):
+            val = np.longdouble(val)
+            if val2 is not None:
+                raise ValueError("Only one value permitted")
+            return val, 0
+
+        def set_jds(self, val, val2):
+            mjd1 = np.float64(np.floor(val))
+            mjd2 = np.float64(val - mjd1)
+            self.jd1, self.jd2 = day_frac(mjd1 + DJM0, mjd2)
+
+        @property
+        def value(self):
+            mjd1, mjd2 = day_frac(self.jd1 - DJM0, self.jd2)
+            return np.longdouble(mjd1) + np.longdouble(mjd2)
+
+    m = 58000.0
+    t = Time(m, format=custom_format_name)
+    t2 = Time(m + 2 * m * np.finfo(np.longdouble).eps, format=custom_format_name)
+    assert t != t2
+    assert isinstance(getattr(t, custom_format_name), np.longdouble)
+    assert getattr(t, custom_format_name) != getattr(t2, custom_format_name)
+
+
+def test_mjd_unit_validation():
```

Comment:
These two unit conversion tests are unrelated to the code change, but presuming they are just to improve coverage then fine to leave them.